server/models/Profile.js

```diff
@@ -0,0 +1,132 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcryptjs");
+
+const profileSchema = new mongoose.Schema({
+  user_username: {
+    type: String,
+    required: true,
+  },
+  first_name: {
+    type: String,
+    required: true,
+  },
+  last_name: {
+    type: String,
+    required: true,
+  },
+  availability: {
+    begin: Date,
+    end: Date,
+    days: {
+      mon: {
+        from: {
```

Comment:
Alternatively, you could use a subdocument to set ranges here([docs](https://mongoosejs.com/docs/subdocs.html#subdocuments))